### PR TITLE
[ART-1176] refactoring elliott find_builds_cli.py bug fixes and removing branch option and product_version

### PR DESCRIPTION
**Pull request by shiywang**
This is the same PR with https://github.com/openshift/elliott/pull/103, but that one had some problems with CI pip install yaml package.

### CODE REVIEW

File: elliottlib/cli/find_builds_cli.py [hunk 1]
```diff
@@ -184,46 +191,27 @@ def _fetch_builds_by_kind_image(runtime, tag_pv_map, session):
...
-    return [
-        b for b in results
-        if not b.attached_to_open_erratum
-        # filter out 'openshift-enterprise-base-container' since it's not needed in advisory
-        if 'openshift-enterprise-base-container' not in b.nvr
-    ]
+    nvrs = []
+    for meta in image_tuples:
+        nvrs.append('{}-{}-{}'.format(meta[0], meta[1], meta[2]))
+    return nvrs
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: Here you lose the product version values.

File: elliottlib/cli/find_builds_cli.py [hunk 2]
```diff
@@ -91,33 +87,42 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
...
+        click.echo('Manually verifying the builds exist')
+        unshipped_nvrs = builds
     elif from_diff:
-        unshipped_builds = _fetch_builds_from_diff(from_diff[0], from_diff[1], product_version, session)
+        green_print('Fetching changed images between payloads...')
+        unshipped_nvrs = elliottlib.openshiftclient.get_build_list(from_diff[0], from_diff[1])
     else:
         if kind == 'image':
-            unshipped_builds = _fetch_builds_by_kind_image(runtime, tag_pv_map, session)
+            unshipped_nvrs = _fetch_builds_by_kind_image(runtime, tag_pv_map)
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: Converting build objects to nvr strings then fetch building objects again is duplicate effort and time consuming.

File: elliottlib/cli/find_builds_cli.py [hunk 3]
```diff
@@ -127,6 +132,20 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
...
+    product_version = ""
+    for tag in brew_session.listTags(nvr):
+        tag_name = tag.get('name')
+        product_version = product_version_map.get(tag_name, "")
+        if product_version:
+            return product_version
+
+
+def _nvrs_to_builds(nvr, product_version_map):
+    pv = _get_product_version(nvr, product_version_map)
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: Here you get the product version values back.

File: elliottlib/cli/find_builds_cli.py [hunk 4]
```diff
@@ -127,6 +132,20 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
             click.echo(' ' + b.nvr)
 
 
+def _get_product_version(nvr, product_version_map, brew_session=koji.ClientSession(constants.BREW_HUB)):
+    product_version = ""
+    for tag in brew_session.listTags(nvr):
+        tag_name = tag.get('name')
+        product_version = product_version_map.get(tag_name, "")
+        if product_version:
+            return product_version
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: It is possible that one brew build is associated with multiple candidate tags.

File: elliottlib/cli/find_builds_cli.py [hunk 5]
```diff
@@ -142,30 +161,18 @@ def _determine_errata_info(runtime):
     if not runtime.branch:
         raise ElliottFatalError('Need to specify a branch either in group.yml or with --branch option')
     base_tag = runtime.branch
-
     et_data = runtime.gitdata.load_data(key='erratatool').data
-    product_version = override_product_version(et_data.get('product_version'), base_tag)
     tag_pv_map = et_data.get('brew_tag_product_version_mapping')
-    return base_tag, product_version, tag_pv_map
+    return base_tag, tag_pv_map
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: I don't think there is any need to use `runtime.branch` after we have the tag_product_version_mapping,

**Comment by vfreex** [comment]
@shiywang This actually does more than ART-1176. Please also update the title and description of this PR and commit message to reflect what you are doing.

File: elliottlib/cli/find_builds_cli.py (hunk 4 above)
Comment (shiywang) [reply elliottlib/cli/find_builds_cli.py]: this function is no longer needed, I've already removed it.

File: elliottlib/cli/find_builds_cli.py (hunk 5 above)
Comment (shiywang) [reply elliottlib/cli/find_builds_cli.py]: yeah, removed.

File: elliottlib/cli/find_builds_cli.py [hunk 6]
```diff
@@ -127,103 +133,78 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
...
+            base_tag = tag[:-10]
+        else:
+            red_print("key of brew_tag_product_version_mapping in erratatool.yml must be candidate\n")
+            continue
+        candidates = elliottlib.brew.find_unshipped_build_candidates(
+            base_tag,
+            tag_pv_map[tag],
+            kind='rpm')
+        pbar_header('Gathering additional information: ', 'Brew buildinfo is required to continue', candidates)
+        rpm_tuple.extend(_gen_nvrp_tuples('nvr', candidates, tag_pv_map))
```
Comment (vfreex) [inline elliottlib/cli/find_builds_cli.py]: `find_unshipped_build_candidates` calls brew list-tagged CLI then `_gen_nvrp_tuples` also calls Brew APIs. Those Brew CLI/API calls can be combined for better performance but can be done in separated cards.

**Pull request merged**